lob.go

```diff
@@ -264,16 +264,38 @@ func (dl *DirectLob) ReadAt(p []byte, offset int64) (int, error) {
 // WriteAt writes p starting at offset.
 func (dl *DirectLob) WriteAt(p []byte, offset int64) (int, error) {
 	if !dl.opened {
-		//fmt.Printf("open %p\n", lob)
+		// fmt.Printf("open %p\n", lob)
 		if C.dpiLob_openResource(dl.dpiLob) == C.DPI_FAILURE {
 			return 0, errors.Errorf("openResources(%p): %w", dl.dpiLob, dl.conn.getError())
 		}
 		dl.opened = true
 	}
-
+	
 	n := C.uint64_t(len(p))
 	if C.dpiLob_writeBytes(dl.dpiLob, C.uint64_t(offset)+1, (*C.char)(unsafe.Pointer(&p[0])), n) == C.DPI_FAILURE {
 		return int(n), errors.Errorf("writeBytes: %w", dl.conn.getError())
 	}
 	return int(n), nil
 }
+
+// GetFileName Return []string consisting of the directory alias and file name for a BFILE type LOB.
+func (dl *DirectLob) GetFileName() ([]string, error) {
+	var directoryAliasLength, fileNameLength C.uint32_t
+	var directoryAlias, fileName *C.char
+	var dir, file string
+	if C.dpiLob_getDirectoryAndFileName(dl.dpiLob,
+		&directoryAlias,
+		&directoryAliasLength,
+		&fileName,
+		&fileNameLength) == C.DPI_FAILURE {
+		err := dl.conn.getError()
+		// C.dpiStmt_release(st.dpiStmt)
+		return nil, errors.Errorf("GetFileName: %w", err)
+	}
+	dir = C.GoStringN(directoryAlias, C.int(directoryAliasLength))
+	file = C.GoStringN(fileName, C.int(fileNameLength))
+	if dir == "" && file == "" {
+		return nil, errors.Errorf("GetFileName error Directory or FileName is null")
```

Comment:
Do we really want to check for this?
Can't this happen in normal circumstances?
Is this an error of the caller, or the database?